### PR TITLE
Wlt 600 delete ba

### PR DESCRIPTION
**Pull request by pol9kov**
**- What I did**
Fixed the bug with createMember method. Burn addreess wasn't deleted
**- How I did it**
Edited root domain method GetBurnAddress.  
**- How to verify it**
Launch TestCreateMemberWhenNoBurnAddressesLeft.
**- Description for the changelog**

### CODE REVIEW

File: application/contract/rootdomain/rootdomain.go [hunk 1]
```diff
@@ -97,12 +97,15 @@ func (rd *RootDomain) AddBurnAddress(burnAddress string) error {
 	return nil
 }
 
-func (rd RootDomain) GetBurnAddress() (string, error) {
+func (rd *RootDomain) GetBurnAddress() (result string, err error) {
```
Comment (megge-dream) [inline application/contract/rootdomain/rootdomain.go]: Functions and methods should not use named return parameters (our code style, 3.2)

File: application/contract/rootdomain/rootdomain.go [hunk 2]
```diff
@@ -97,12 +97,15 @@ func (rd *RootDomain) AddBurnAddress(burnAddress string) error {
...
 	if len(rd.FreeBurnAddresses) == 0 {
-		return "", fmt.Errorf("no more burn address left")
+		return "", fmt.Errorf("no more burn addresses left")
 	}
 
-	return rd.FreeBurnAddresses[0], nil
+	result = rd.FreeBurnAddresses[0]
+	rd.FreeBurnAddresses = rd.FreeBurnAddresses[1:]
```
Comment (megge-dream) [inline application/contract/rootdomain/rootdomain.go]: What if we get error while adding new member to maps? Should we return burn address to list of free addresses in that case?

Comment (pol9kov) [reply application/contract/rootdomain/rootdomain.go]: Yes. I created rollBack func in this PR https://github.com/insolar/insolar/pull/1966

File: application/contract/rootdomain/rootdomain.go (hunk 1 above)
Comment (pol9kov) [reply application/contract/rootdomain/rootdomain.go]: In this case it is not get method in standard meaning.

Comment (pol9kov) [reply application/contract/rootdomain/rootdomain.go]: RootDomain doesn't have field burnAddress. Instead in this method we get one burn address from list.

Comment (megge-dream) [reply application/contract/rootdomain/rootdomain.go]: I'm talking about return parameters, not get prefix in method name

**Pull request merged**